src/core/app-template.cc

```diff
@@ -118,11 +119,25 @@ const app_template::seastar_options& app_template::options() const {
 app_template::configuration_reader app_template::get_default_configuration_reader() {
     return [this] (bpo::variables_map& configuration) {
         auto home = std::getenv("HOME");
-        if (home) {
-            std::ifstream ifs(std::string(home) + "/.config/seastar/seastar.conf");
-            if (ifs) {
-                bpo::store(bpo::parse_config_file(ifs, _opts_conf_file), configuration);
+
+        std::string config_file = _opts.config_file.get_value();
+        if (config_file.empty()) {
+            throw std::runtime_error("Empty 'config-file'");
+        }
+        if (config_file[0] == '~') {
```

Comment:
~ expands to $HOME only if not followed by a word. ~user = user's home.

Reply:
Is it needed here? Let bash perform the expansion, and if it's not run from a bash script, then the user should be explict.

Reply:
Won't it just work? bash will expand it.

```sh
$ /bin/echo ~/foo
/home/avi/foo
```

Reply:
Yes, provided it's run from bash without quotes
```
$ /bin/echo '~/'
~/
```
and without fancy variabling
```
$ FOO='~/'; echo ${FOO}
~/
```

But it's indeed doesn't worth the code to parse. I'll drop it